### PR TITLE
Added more information on diff() modifier

### PR DESCRIPTION
**Pull request by estherk15**
### What does this PR do?
- Add information on the diff() modifier and update the Functions Overview page with updated UI image.
- Removed the images for examples as they are out of date with the UI
- Changed the name of the modifiers to match the UI:
image

### Motivation
[DOCS-4347](https://datadoghq.atlassian.net/browse/DOCS-4347)

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/dashboards/functions/_index.md [hunk 1]
```diff
@@ -23,18 +23,15 @@ aliases:
 
 ## Overview
 
-Functions can be applied to your queries by clicking `+` icon in the graphing editor. Most of the functions are applied as the last step (after [time][1] and [space aggregation][2]).
+Functions can modify how the results of a metric query are returned for visualizations. Most functions are applied after the results of the metric query are returned, but functions can also change the parameters before the query is made. For example, the Rollup function changes the time aggregation of a query before the results are returned. Alternatively, arithmetic functions apply changes to the returned results of the metric query. See the [Metrics][3] page to learn more about querying metrics. To learn more about the different functions, see the [function types](#function-types).
```
Comment (alai97) [inline content/en/dashboards/functions/_index.md:26]:
```suggestion
Functions can modify how the results of a metric query are returned for visualizations. Most functions are applied after the results of the metric query are returned, but functions can also change the parameters before the query is made. 

For example, the Rollup function changes the time aggregation of a query before the results are returned. Alternatively, arithmetic functions apply changes to the returned results of the metric query. For more information about querying metrics, see [Metrics][3]. To learn more about the different functions, see the [function types](#function-types) section.
```

File: content/en/dashboards/functions/_index.md [hunk 2]
```diff
@@ -54,3 +51,4 @@ Here is an example of how to apply a timeshift function on your error logs to co
```
Comment (alai97) [inline content/en/dashboards/functions/_index.md:51]: Should we add a Further Reading section here with some links defined in the front matter too?

File: content/en/dashboards/functions/rate.md [hunk 3]
```diff
@@ -3,6 +3,10 @@ title: Rate
 kind: documentation
 aliases:
     - /graphing/functions/rate/
+further_reading:
+- link: "/monitors/guide/alert-on-no-change-in-value/"
+  tag: "Guide"
```
Comment (alai97) [inline content/en/dashboards/functions/rate.md:8]:
```suggestion
  tag: "Documentation"
```

File: content/en/dashboards/functions/rate.md [hunk 4]
```diff
@@ -3,6 +3,10 @@ title: Rate
...
+  text: "Alert on no change in value"
 ---
```
Comment (alai97) [inline content/en/dashboards/functions/rate.md:11]: Can we add an Overview section that says "this page describes the rate function"?

**Review by alai97 — CHANGES_REQUESTED** [review]
Looks good, some suggestions!

File: content/en/dashboards/functions/rate.md (hunk 3 above)
Comment (estherk15) [reply content/en/dashboards/functions/rate.md:8]: The kind for this page is listed as "guide", do we default to documentation for tags?

File: content/en/dashboards/functions/rate.md (hunk 4 above)
Comment (estherk15) [reply content/en/dashboards/functions/rate.md:11]: Thanks Austin, that makes sense. I'm planning on restructuring all the function pages. When I outline a template for them, I'll add an overview section for each page! Or would you recommend adding the section anyway?

File: content/en/dashboards/functions/rate.md (hunk 3 above)
Comment (alai97) [reply content/en/dashboards/functions/rate.md:8]: Yep, we don't explicitly state "Guide" or "FAQ", just "Documentation" usually!

**Review by alai97 — APPROVED** [review]

**Pull request merged**